pkg/recipes/driver/types.go

```diff
@@ -42,6 +42,13 @@ type Driver interface {
 	GetRecipeMetadata(ctx context.Context, opts BaseOptions) (map[string]any, error)
 }
 
+type DriverWithSecrets interface {
+	Driver
+
+	// FindSecretIDs gets the secret ID references from environment definition.
+	FindSecretIDs(ctx context.Context, config recipes.Configuration, definition recipes.EnvironmentDefinition) (string, error)
```

Comment:
Should this return `([]string, error)`?

What would happen in the future if a driver needed to read multiple secrets?

Reply:
currently we are keeping it return module secret. But in the future when we have provider secrets etc, return type will be map because we may need to specify secret kind then ex: module secret or provider secret. It will be discussed in detail during the design for provider secrets.